src/rules.jl

```diff
@@ -84,7 +84,7 @@ _abs_deriv(x) = signbit(x) ? -one(x) : one(x)
 @define_diffrule Base.:*(x, y) = :( $y                 ), :(  $x                )
 @define_diffrule Base.:/(x, y) = :( inv($y)            ), :( -($x / $y / $y)    )
 @define_diffrule Base.:\(x, y) = :( -($y / $x / $x)    ), :( inv($x)            )
-@define_diffrule Base.:^(x, y) = :( $y * ($x^($y - 1)) ), :(  ($x^$y) * log($x) )
+@define_diffrule Base.:^(x, y) = :( $y * ($x^($y - 1)) ), :( $x > 0 ? ($x^$y) * log($x) : NaN )
```

Comment:
Since this package depends on NaNMath.jl and
```
julia> NaNMath.log(2.0f0)
0.6931472f0

julia> NaNMath.log(-2.0f0)
NaN32
```
the obvious thing to do is
```suggestion
@define_diffrule Base.:^(x, y) = :( $y * ($x^($y - 1)) ), :(  ($x^$y) *  $(NaNMath.log)($x) )
```

Reply:
Bad idea. It depends on the package only to fix a version (I presume) but does not load it. So this doesn't work.